src/timedate/timedated.c

```diff
@@ -800,6 +860,11 @@ static int method_set_ntp(sd_bus_message *m, void *userdata, sd_bus_error *error
         if (r == 0)
                 return 1;
 
+        /* This method may be called frequently. Forget the previous job if it has not completed yet.
+         * In that case, the active state obtained by context_update_ntp_status() may no be reliable. */
+        ignore_active_state = c->slot_job_removed;
+        c->slot_job_removed = sd_bus_slot_unref(c->slot_job_removed);
```

Comment:
So... is this check really necessary? Wouldn't it be simpler and more reliable to just send the job start/stop request always? Imagine the following scenario:
1. the service is not running
2. somebody calls enable through `timedated`, `timedated` queues a start job
3. the start job finishes
4. a stop job is queued through `systemctl stop`
5. somebody calls enable through `timedated`, `timedated` queues a start job

Even if 4 and 5 are always in that order, depending on whether 3 is earlier or later, the outcome will be different (in one case a new start job will be queued, in the other not).

My thinking is that pid1 is the only place that really know the current state, and the only cost we have from forwarding all the enable/disable calls to it is a dbus call, which is not expensive.

Reply:
That is a very good question. While I update this, I noticed that. So, can I confirm your suggestion? You mean that we should set in some sense that `ignore_active_state` always true?

I will test the racy conditions later and may update this. Thank you so much for giving comments so many times.